yt/loaders.py

```diff
@@ -151,8 +155,26 @@ def load_simulation(fn, simulation_type, find_outputs=False):
 # --- Loaders for generic ("stream") data ---
 
 
+def _sanitize_axis_order_args(
+    geometry: Union[str, Tuple[str, AxisOrder]], axis_order: Optional[AxisOrder]
+) -> Tuple[str, Optional[AxisOrder]]:
+    # this entire function should be removed at the end of its deprecation cycle
+    geometry_str: str
+    if isinstance(geometry, tuple):
+        issue_deprecation_warning(
+            f"Received a tuple as {geometry=}\n"
+            "Use the `axis_order` argument instead.",
+            since="4.2",
+            stacklevel=4,
+        )
+        geometry_str, axis_order = geometry
+    else:
+        geometry_str = geometry
+    return geometry_str, axis_order
+
+
 def load_uniform_grid(
-    data,
+    data: Dict[AnyFieldKey, np.ndarray],
```

Comment:
I'm planning to review the whole PR in more detail... but a quick comment: load_uniform_grid can accept data defined via callables similar to load_amr_grids. It's described in the docstring here, but looks like we don't actually have a unit test with that... I should add one...

Reply:
Thanks for pointing this out. So I guess this is #4255 ?

Reply:
So here's a diff between this PR's branch and an attempt to add type hints: https://gist.github.com/chrishavlin/4c8e55179299e60d88dc5129e32cc676 

I ran into a few issues: trying to use `AMRGridPatch` in `yt._typing` resulted in a circular import. And then there was one spot I couldn't get passing without a much bigger refactor, so I gave up and added an ignore. As it is, the diff is minor enough that it could be easily included here. I could submit a follow up PR to annotate `load_uniform_grid` and work on it some more (in which case I think the type hint on this line could just be removed).

Reply:
As far as I am concerned it doesn't need to be a follow up, I think it could be its own thing and take priority over the present PR. We can probably avoid circular imports by putting the definition in the module where it's used.

Reply:
just looked at it again, and it actually makes use of some of your typing improvements from this PR, so it'd actually be blocked by this PR anyway. So I think I'll just give this PR a full review before starting a new one based on this branch...

Reply:
Whatever works for you :)